### PR TITLE
Phantom js

### PR DESCRIPTION
**Pull request by allanderek**
Hey, just a minor update to some of the coffee script code. You can read the commits but basically ensures that we can detect what flavour (empty, black or white) a board position is by looking at the class of the td element, rather than checking the image filename. This makes the actual counting a touch easier.

### CODE REVIEW

File: app/coffee/tests/browser.coffee [hunk 1]
```diff
@@ -40,17 +40,15 @@ casper.test.begin "Game interface", 23, (test) ->
...
-      'black': countImgsWith 'b.gif'
-      'white': countImgsWith 'w.gif'
+    counts = casper.evaluate () ->
+        go_points = $('table.goban .gopoint').length
+        black_stones = $('.blackstone').length
+        white_stones = $('.whitestone').length
+        counts =
+          'empty': go_points - (black_stones + white_stones)
+          'black': black_stones
+          'white': white_stones
```
Comment (karlorg) [inline app/coffee/tests/browser.coffee]: don't forget that CS style for variable names is camelCase.

File: app/main.py [hunk 2]
```diff
@@ -316,7 +316,7 @@ def get_goban_from_moves(moves, setup_stones=None):
         setup_stones = []
     goban = [[dict(
         img=IMG_PATH_EMPTY,
-        classes='row-{row} col-{col}'.format(row=str(j), col=str(i))
+        classes='gopoint row-{row} col-{col}'.format(row=str(j), col=str(i))
```
Comment (karlorg) [inline app/main.py]: maybe worth adding a line to TestGetGobanFromMoves.assert_point to assert that the class `gopoint` is appearing in the output?  Then if it goes missing, the test failures will point to this function instead of the CS tests mysteriously miscounting the empty points.

**Comment by karlorg** [comment]
On 17.03.2015 16:34, allanderek wrote:
> Hey, just a minor update to some of the coffee script code. You can read
> the commits but basically ensures that we can detect what flavour
> (empty, black or white) a board position is by looking at the class of
> the td element, rather than checking the image filename. This makes the
> actual counting a touch easier.

Cool.  Couple of line notes, see what you think, and let me know when 
you've either made changes or decided not to bother, and then I'll pull it.

BTW try not to change the game view test too much at this point as I 
have lots of changes to that one in my go rules local branch, so there's 
a danger of making changes that are now irrelevant and/or confuse the 
merge system.

File: app/coffee/tests/browser.coffee [hunk 3]
```diff
@@ -40,25 +40,25 @@ casper.test.begin "Game interface", 23, (test) ->
...
-  test.assertPointMatches = (x, y, regex) ->
-    imgSrc = casper.evaluate ((imgSel) -> $(imgSel).attr 'src'),
-                             pointSelector(x, y) + " img"
-    test.assertMatch imgSrc, regex
+  test.assertPointIsBlack = (x,y) ->
+    test.assertExists pointSelector(x,y) + ".blackstone"
+  test.assertPointIsWhite = (x,y) ->
+    test.assertExists pointSelector(x,y) + ".whitestone"
+  test.assertPointIsEmpty = (x,y) ->
+    test.assertExists pointSelector(x,y) + ".nostone"
```
Comment (karlorg) [inline app/coffee/tests/browser.coffee]: Will this `assertPointIsEmpty` work for empty points that have not been modified since the page was loaded?  I don't see anything in `main.py` that adds the `nostone` class

Comment (allanderek) [inline app/coffee/tests/browser.coffee]: Okay yeah perfect.
